core/jvm/src/main/scala/zio/sql/typetag.scala

```diff
@@ -1,17 +1,20 @@
 package zio.sql
 
+import java.sql.ResultSet
 import java.time._
 import java.util.UUID
-
 import zio.Chunk
 
 trait TypeTagModule {
 
-  type TypeTagExtension[+A]
+  type TypeTagExtension[+A] <: Tag[A]
```

Comment:
hint: Tag[A] with Decodeable[E, A]? not sure if we need the E, won't this always be a DecodeError?

Reply:
yeah that's a good shout out.